### PR TITLE
Some vision fixes

### LINKED ISSUE #15282: Weird Overlays in Cryo
sybil mime city oscar 2016-02-07 132114
This was also overlayed over the blind/sleeping darkness.  Don't have any idea what could be causing it, apparently nobody else but me could see it.

### LINKED ISSUE #15304: grabbing a meat spiked person gimps your movement speed
Try to use a grab on someone who is meat spiked and you will be moving at a walking pace forever. Have had to respawn people to fix it.

### LINKED ISSUE #15307: Borgs are immune to flashing
You get the "overloads sensor" message, but no blind or stun as the cyborg.

### LINKED ISSUE #15308: Cryo vision is weird
You can't see if it's closed and turned on if you're inside of it which is very strange.

### LINKED ISSUE #15309: Cloning/cryo bug
Upon getting out of either you never fall down, you remain standing but asleep.

### LINKED ISSUE #15310: Borgs never get unstunned once stunned
I got EM rifled, was perma unconscious

### PR DESCRIPTION
**Pull request by phil235**
- Fixes unconscious mob standing still when removed from cryo cell, closet, containers. Fixes #15309
- Fixes mobs inside cryo cells getting pipes overlay(ventcrawl). Fixes #15282 Fixes #15308
- Ventcrawler can no longer move in or out ofthe cryo cell, which fixes ventcrawler going into cryo cell and getting stuck in it.
- Being inside a cryo cell now give you a nearsight overlay (same as closet).
- Brains no longer get the red circles overlay when damaged.
- MMI now properly updates its icon when the brain mob dies.
- Fixes getting stuck with a grab inside you when grabbing someone buckled. Fixes #15304
- Silicon stunned, weakened and paralysis is now properly updated. Fixes #15307 Fixes #15310

### CODE REVIEW

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 1]
```diff
@@ -532,13 +532,14 @@
...
 		icon_state = "alarmp"
 		return
-	switch(max(danger_level, alarm_area.atmosalm))
-		if (0)
-			src.icon_state = "alarm0"
-		if (1)
-			src.icon_state = "alarm2" //yes, alarm2 is yellow alarm
-		if (2)
-			src.icon_state = "alarm1"
+	if(alarm_area)
```
Comment (neersighted) [inline code/modules/atmospherics/machinery/airalarm.dm]: Stealth fix... I like it.

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 2]
```diff
@@ -256,3 +254,15 @@
...
 	update_icon()
+
+/obj/machinery/atmospherics/components/unary/cryo_cell/update_remote_sight(mob/living/user)
+	return //we don't see the pipe network while inside cryo.
+
+/obj/machinery/atmospherics/components/unary/cryo_cell/get_remote_view_fullscreens(mob/user)
+	user.overlay_fullscreen("remote_view", /obj/screen/fullscreen/impaired, 1)
+
+/obj/machinery/atmospherics/components/unary/cryo_cell/can_crawl_through()
+	return //can ventcrawl in or out of cryo.
```
Comment (neersighted) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: You mean can't?

File: code/modules/mob/living/living_defense.dm [hunk 3]
```diff
@@ -184,13 +184,14 @@
 
 	add_logs(user, src, "grabbed", addition="passively")
 
-	var/obj/item/weapon/grab/G = new /obj/item/weapon/grab(user, src)
+	if(anchored || !user.Adjacent(src))
```
Comment (neersighted) [inline code/modules/mob/living/living_defense.dm]: Could be `Adjacent(user)`

File: code/modules/atmospherics/machinery/airalarm.dm (hunk 1 above)
Comment (phil235) [inline code/modules/atmospherics/machinery/airalarm.dm]: shit, not a fix. I didn't meant to fix that. It's just that I had to make a temporary fix when doing my testing, otherwises this stuff creates so many runtimes that it doesn't show the next potential runtimes.

File: code/modules/mob/living/living_defense.dm [hunk 4]
```diff
@@ -184,13 +184,14 @@
...
+		return 0
 	if(buckled)
 		user << "<span class='warning'>You cannot grab [src], \he is buckled in!</span>"
 		return 0
-	if(!G)	//the grab will delete itself in New if src is anchored
-		return 0
-	user.put_in_active_hand(G)
+	var/obj/item/weapon/grab/G = new /obj/item/weapon/grab(user, src)
+	if(!user.put_in_active_hand(G)) //if we can't put the grab in our hand for some reason, we delete it.
+		qdel(G)
```
Comment (neersighted) [inline code/modules/mob/living/living_defense.dm]: `qdel(G)`?

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm (hunk 2 above)
Comment (phil235) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: derp

File: code/modules/mob/living/living_defense.dm (hunk 4 above)
Comment (phil235) [inline code/modules/mob/living/living_defense.dm]: See https://github.com/tgstation/-tg-station/pull/15327/files#diff-0d58be00f177298faed0e648e735056eL27
If we somehow didn't manage to put the grab into our hand, we delete it.

Comment (neersighted) [inline code/modules/mob/living/living_defense.dm]: Oh, the github diff viewer goof'd and it looked like `(G) qdel` to me.

**Pull request merged**